### DISCUSSION TITLE
Preview before save

### DISCUSSION THREAD

**Discussion by alexbechmann** [payloadcms/payload#326] [Feature Requests & Ideas]
I would love to see the ability to preview before you commit to saving.

Maybe the "Preview" button should be renamed to "View" and a new preview button should implement this feature.

I see two options:

- Developer can provide a preview react component that receives all of the current input data as props. Clicking preview would open up a modal with the component on it.
- Clicking preview will send a POST request (in a new tab?) with current data (as if it had been saved) to a URL. Then a developer can implement a post route that renders a view on the frontend using that data.

**Comment by jmikrut** [comment]
Hey @alexbechmann — I totally agree with your wishes here.

Although I do like the options you've provided, we do have some ideas in flight already here. Specifically to do with these two items on our roadmap:

- [Revision History](https://github.com/payloadcms/payload/discussions/292)
- [Autosave Draft Changes](https://github.com/payloadcms/payload/discussions/295)

With these two features being added by end of year, we'll be able to make that existing Preview button much more functional. Instead of loading the most recent "published" draft, it'll instead load the most recent draft. That will solve the issue.

What do you think?

**Reply by alexbechmann** [reply]
Hi @jmikrut, Yeah that should work fine! Glad to hear that it's in progress :)
